codestyle-guide.md

```diff
@@ -0,0 +1,90 @@
+# Grafana Plugins Code Style Draft
+
+## Repo name
+It's a good practice to use plugin repo name same as plugin id.  
+Plugin id has following structure:
+```
+(username|company)-(plugin name)-(datasource|app|panel)
+```
+
+Examples:
+```
+raintank-worldping-app
+grafana-simple-json-datasource
+grafana-piechart-panel
+mtanda-histogram-panel
+```
+
+## Directories structure
+
+```
+johnnyb-awesome-datasource
+|-- dist
+|-- spec
+|   |-- datasource_spec.js
+|   |-- query_ctrl_spec.js
+|   |-- test-main.js
+|-- src
+|   |-- img
+|   |   |-- logo.svg
+|   |-- partials
+|   |   |-- annotations.editor.html
+|   |   |-- config.html
+|   |   |-- query.editor.html
+|   |-- datasource.js
+|   |-- module.js
+|   |-- plugin.json
+|   |-- query_ctrl.js
+|-- Gruntfile.js
+|-- LICENSE
+|-- package.json
+|-- README.md
+```
+
```

Comment:
We might want to note the option of putting a README.md for grafana/grafana.net under src/, that can be different from the github README.md in the repo root.  Also adding example screenshots might encourage people to provide them :)